File: java/src/test/java/com/kcl/TestingTest.java
package com.kcl;

import com.kcl.api.API;
import com.kcl.api.Spec.TestArgs;
import com.kcl.api.Spec.TestResult;

import org.junit.Assert;
import org.junit.Test;

public class TestingTest {
    @Test
    public void testTestingApi() throws Exception {
        API apiInstance = new API();
        TestArgs args = TestArgs.newBuilder().addPkgList("./src/test_data/testing/...").build();
        TestResult result = apiInstance.test(args);
        Assert.assertEquals(result.getInfoCount(), 2);
    }
}

File: kotlin/src/test/kotlin/com/kcl/TestingTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.testArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class TestingTest {
    companion object {
        private const val TEST_PKG = "./src/test_data/testing/..."
    }

    @Test
    fun testTestingApi() {
        val args = testArgs {
            pkgList += TEST_PKG
        }
        val api = API()
        val result = api.test(args)
        assertEquals(result.infoCount, 2)
    }
}
